lib/rubocop/rspec/expect_offense.rb

```diff
@@ -113,6 +113,8 @@ def format_offense(source, **replacements)
 
       # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
       def expect_offense(source, file = nil, severity: nil, **replacements)
+        source = "#{source.join("\n")}\n" if source.is_a?(Array)
```

Comment:
I'm :-1: on this, I much prefer if we use Heredocs instead of arrays of lines which aren't nearly as readable. Let me know if I missed a use-case.

Reply:
Now I agree that it looks more like a hacky workaround than a good solution (because it is :) ).          
As to why I did this: for trailing whitespaces. 
In cases like this - it's not clearly seen are there whitespaces, how much of them are there.       
Also some folks have removing trailing whitespace done automatically by the IDE on save - so one must be careful with these specs.
```ruby
    expect_offense(<<~RUBY)
        
      ^^ Trailing whitespace detected.\n
    RUBY
```
But later I've found a helper method `trailing_whitespace` in (`/spec/support/misc_helper.rb`) - I think it will solve the problem like this: 
 ```ruby
    expect_offense(<<~RUBY)
      #{trailing_whitespace}  
      ^^ Trailing whitespace detected.\n
    RUBY
```    
What do you think ?

Reply:
Indeed. BTW RuboCop will flag and auto-correct trailing whitespaces in heredocs now too.